gtsam/hybrid/HybridConditional.cpp

```diff
@@ -169,4 +169,25 @@ double HybridConditional::evaluate(const HybridValues &values) const {
   return std::exp(logProbability(values));
 }
 
+/* ************************************************************************ */
+void HybridConditional::removeDiscreteModes(const DiscreteValues &given) {
```

Comment:
Could be moved to DiscreteConditional?

Reply:
Moved it and changed the for loop to a `std::for_each` to be more idiomatic.